test-data/unit/check-inference.test

```diff
@@ -1392,8 +1392,9 @@ class A:
         self.x = [] # E: Need type annotation for 'x'
 
 class B(A):
+    # TODO?: This error is kind of a false positive, unfortunately
     @property
-    def x(self) -> List[int]:
+    def x(self) -> List[int]:  # E: Signature of "x" incompatible with supertype "A"
```

Comment:
Perhaps related to #3260?

Reply:
I think the issue is that `x` in the parent is given the type `List[Any]` which isn't compatible with `List[int]`?